File: Python/169-Majority-Element.py
class Solution(object):
    def majorityElement(self, nums):
        cache = {}
        length = len(nums)
        for num in nums:
            if num in cache:
                cache[num] += 1
            else:
                cache[num] = 1
            if cache[num] > length / 2:
                return num

File: JavaScript/169-majority-element.js
let majorityElement = nums =>
  nums.reduce(
    ([cdd, cnt], n) => [
      (cnt == 0 && (cdd = n), cdd),
      cdd == n ? cnt + 1 : cnt - 1,
    ],
    [0, 0],
  )[0]
